source/distributed/tria.cc

```diff
@@ -3272,6 +3470,11 @@ namespace parallel
           // fix all the flags to make sure we have a consistent mesh
           this->prepare_coarsening_and_refinement ();
 
+          // enforce 2:1 mesh balance over periodic boundaries
+          if (this->smooth_grid & dealii::Triangulation<dim,spacedim>::limit_level_difference_at_vertices)
+            enforce_mesh_balance_over_periodic_boundaries(*this,
```

Comment:
Are we sure this produces the correct result with a single iteration? Can you add a check (maybe wrapped in ``#ifdef DEBUG``) that runs ``enforce_mesh_balance_over_periodic_boundaries`` after ``execute_coarsening_and_refinement`` below to check if any cells will be flagged?